plutus-contract/src/Plutus/Contract/Wallet.hs

```diff
@@ -168,7 +174,7 @@ addInputs mp pk vl tx = do
 
         addTxIns  =
             let ins = Set.fromList (Tx.pubKeyTxIn . fst <$> spend)
-            in over Tx.inputs (Set.union ins)
+            in over Tx.inputsFees (Set.union ins)
```

Comment:
Sometimes the tx inputs that are added here have nothing to do with the fees. For example if just want to include a token in the transaction. In that case the input should be added to `Tx.inputs`. Maybe we should first balance the transaction, putting all new inputs into `Tx.inputs`, and then in a second step move some of the inputs from `Tx.inputs` to `Tx.inputsFees`. Then we also won't need to add the fees to `neg'` (I think)